File: AppointmentApi/CalDate.cs
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;

namespace CalBookApi {
    [BsonIgnoreExtraElements]
    public class CalDate {
        [BsonElement(elementName: "dd")]
        public int dd { get; set; } 
        
        [BsonElement(elementName: "mm")]
        public int mm { get; set; } 

        [BsonElement(elementName: "yyyy")]
        public int yyyy { get; set; } 

        [BsonConstructor]
        public CalDate(int dD, int mM, int yYYY) {
            this.yyyy = yYYY;
            this.mm = mM;
            this.dd = dD;
        }

        public CalDate(string dateCode) {
            this.yyyy = int.Parse(dateCode.Substring(0, 4));
            this.mm = int.Parse(dateCode.Substring(4, 2));
            this.dd = int.Parse(dateCode.Substring(6, 2));
        }

        public string ToDateCode() {
            string dateCode = string.Format("{0}{1}{2}",
                this.yyyy.ToString("D4"),
                this.mm.ToString("D2"),
                this.dd.ToString("D2")
            );

            return dateCode;
        }
    }
}

File: src/app/classes/cal-date.ts
export class CalDate {
    public dd: number = 1;
    public mm: number = 1;
    public yyyy: number = 2000;

    constructor(dateCode: string) {
        this.yyyy = parseInt(dateCode.substring(0, 4), 10);
        this.mm = parseInt(dateCode.substring(4, 6), 10);
        this.dd = parseInt(dateCode.substring(6), 10);
    }

    public toDateCode(): string {
        let ddString = (this.dd < 10) ? (`0${this.dd}`) : `${this.dd}`;
        let mmString = (this.mm < 10) ? (`0${this.mm}`) : `${this.mm}`;
        let yyyyString = `${this.yyyy}`;

        return `${yyyyString}${mmString}${ddString}`;
    }

    public toUsaDateFormat(): string {
        let ddString = (this.dd < 10) ? (`0${this.dd}`) : `${this.dd}`;
        let mmString = (this.mm < 10) ? (`0${this.mm}`) : `${this.mm}`;
        let yyyyString = `${this.yyyy}`;

        return `${mmString}/${ddString}/${yyyyString}`;
    }
}
